substrate/primitives/blockchain/src/backend.rs

```diff
@@ -289,9 +289,14 @@ pub trait Backend<Block: BlockT>:
 					match finalized_chain.iter().rev().nth(distance_from_finalized as usize) {
 						Some(header) => (header.number, header.hash),
 						None => {
-							let header = self.header_metadata(
-								finalized_chain.front().expect("Not empty; qed").parent,
-							)?;
+							let parent = finalized_chain.front().expect("Not empty; qed").parent;
+							let header = match self.header_metadata(parent) {
+								Ok(header) => header,
+								Err(_) => {
+									debug!("Can't find obtain header metatadata: {parent:?}");
+									break;
```

Comment:
Does this mean we never remove one of the blocks from the database?

Reply:
If I understand the code correctly we skip only a missing block.